### PR TITLE
5.3 is done

### PR DESCRIPTION
**Pull request by dmytro-hryhoruk**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -9,6 +15,28 @@
  */
 public class FileWork {
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]: should we use `File` here? Try removing it

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -9,6 +15,28 @@
...
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String result = "";
+            String st;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]: st - isn't very informative name of the variable

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -9,6 +15,28 @@
...
+            while ((st = bufferedReader.readLine()) != null) {
+                String[] singleLine = st
+                        .replaceAll("[^a-zA-Z ]", "")
+                        .split("\\s");
+                for (String words : singleLine) {
+                    if (words.toLowerCase().charAt(0) == 'w') {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]:
```suggestion
                    if (words.toLowerCase().charAt(0).startsWith('w')) {
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -9,6 +15,28 @@
...
+                        result += words.toLowerCase() + " ";
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]: better use ArrayList for saving words than String `result`

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -9,6 +16,30 @@
...
+            }
+            if (result.size() == 0) {
+                return new String[0];
+            }
+            Collections.sort(result);
+            String[] resultArray = new String[result.size()];
+            for (int i = 0; i < result.size(); i++) {
+                resultArray[i] = result.get(i);
+            }
+            return resultArray;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]: try using standard method `result.toArray(...)`

File: src/main/java/core/basesyntax/FileWork.java [hunk 6]
```diff
@@ -9,6 +16,30 @@
...
+        } catch (IOException e) {
+            return new String[0];
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]: move this line after catch

File: src/main/java/core/basesyntax/FileWork.java [hunk 7]
```diff
@@ -9,6 +16,30 @@
...
+                        .split("\\s");
+                for (String words : singleLine) {
+                    if (words.toLowerCase().startsWith("w")) {
+                        result.add(words.toLowerCase());
+                    }
+                }
+            }
+            if (result.size() == 0) {
+                return new String[0];
+            }
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]: we can remove this

File: src/main/java/core/basesyntax/FileWork.java [hunk 8]
```diff
@@ -9,6 +16,30 @@
...
-        return null;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            List<String> result = new ArrayList<>();
+            String fileLine;
+            while ((fileLine = bufferedReader.readLine()) != null) {
+                String[] singleLine = fileLine
+                        .replaceAll("[^a-zA-Z ]", "")
+                        .split("\\s");
+                for (String words : singleLine) {
+                    if (words.toLowerCase().startsWith("w")) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java]: make constant for "w" in class

File: src/main/java/core/basesyntax/FileWork.java [hunk 9]
```diff
@@ -8,7 +15,26 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
+    public static final String W = "w";
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FileWork.java]:
```suggestion
    public static final String FIRST_LETTER = "w";
```

**Pull request closed without merging**